src/pineko/cli/check.py

```diff
@@ -36,54 +38,46 @@ def sub_compatibility(grid_path, operator_path, xif):
         rich.print("[red]Error:[/]", e)
 
 
+class Scale(Enum):
+    """Auxiliary class to list the possible scale variations."""
+
+    REN = auto()
+    FACT = auto()
+
+
 @subcommand.command("scvar")
 @click.argument("grid_path", metavar="PINEAPPL", type=click.Path(exists=True))
 @click.argument(
     "scale",
     metavar="SCALE",
-    type=click.Choice(["ren", "fact"]),
+    type=click.Choice(list(map(lambda x: x.name.upper(), Scale)), case_sensitive=False),
 )
 @click.argument("max_as_order", metavar="AS_ORDER", type=int)
 @click.argument("max_al_order", metavar="AL_ORDER", type=int)
 def sub_scvar(grid_path, scale, max_as_order, max_al_order):
     """Check if PineAPPL grid contains requested scale variations for the requested order."""
     grid = pineappl.grid.Grid.read(grid_path)
     grid.optimize()
-    if scale == "ren":
-        is_ren_as, is_ren_al = check.contains_ren(grid, max_as_order, max_al_order)
-        if is_ren_as:
-            rich.print(
-                "[green]Success:[/] grids contain renormalization scale variations for as"
-            )
-        else:
-            rich.print(
-                "[red]Error:[/] grids do not contain renormalization scale variations for as"
-            )
-        if is_ren_al:
-            rich.print(
-                "[green]Success:[/] grids contain renormalization scale variations for al"
-            )
-        else:
-            rich.print(
-                "[red]Error:[/] grids do not contain renormalization scale variations for al"
-            )
-    elif scale == "fact":
-        is_fact_as, is_fact_al = check.contains_fact(grid, max_as_order, max_al_order)
-        if is_fact_as:
-            rich.print(
-                "[green]Success:[/] grids contain factorization scale variations for as"
-            )
-        else:
-            rich.print(
-                "[red]Error:[/] grids do not contain factorization scale variations for as"
-            )
-        if is_fact_al:
-            rich.print(
-                "[green]Success:[/] grids contain factorization scale variations for al"
-            )
+    success = "[green]Success:[/] grids contain"
+    error = "[red]Error:[/] grids do not contain"
+    scale_variations = {
+        Scale.REN.name: " renormalization scale variations",
+        Scale.FACT.name: " factorization scale variations",
+    }
+    if scale not in scale_variations.keys():
+        raise ValueError("Scale variation to check can be one between ren and fact")
+    sv = scale_variations[scale]
+    funcs = {Scale.REN.name: check.contains_ren, Scale.FACT.name: check.contains_fact}
+    func_to_call = funcs[scale]
+    # Call the function
+    is_sv_as, is_sv_al = func_to_call(grid, max_as_order, max_al_order)
+    conditions = {" for as": is_sv_as, " for al": is_sv_al}
```

Comment:
I would define an `Enum` also here: let's start limit dictionaries when not required.

`dict` is very little transparent for the code: you won't get autocompletion or static check, and it is mutable also when you don't want it.

Reply:
Ok I believe for the other case I have done what you suggested. However, in this case, it is not clear to me how to use enum (and actually how to avoid the dictionary).

Reply:
1. You are using the `dict` only to iterate on them
    - and, by the way, if use the value as well, do not iterate on `.keys()`, and access for the value, just iterate on `.items()`
    - moreover, if you iterate on a dictionary is already iterating on its keys, so calling the `.keys()` method is not making any difference
2. you are creating the keys only to write them later, but you see that there is already redundancy: the `" for "` part is repeated twice, you can just drop it and put that below

At this point, you can also avoid the mapping and just iterate over the couples, i.e. start defining an `Enum`:

```py
CouplingInfo = namedtuple("CouplingInfo", ["descr", "theory"])

class Coupling(Enum):
    as = CouplingInfo("strong", "QCD")
    al = CouplingInfo("electromagnetic", "QED")
```

Reply:
Most likely this class can also be lifted somewhere else (later on), since it is a rather general definition.